### ISSUE TITLE
reportUnnecessaryComparison regression in 1.1.286

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#4383]
Hello,

Seems to me this is a regression since 1.1.286. Maybe a side effect of fixing https://github.com/microsoft/pyright/issues/4040 ?

```python
# pyright: strict

from __future__ import annotations

count = 0

for x in range(1000):
    for y in range(10):
        count += 1

if count == 5:
    ...
```

This will log an error `error: Condition will always evaluate to False since the types "Literal[0]" and "Literal[5]" have no overlap (reportUnnecessaryComparison)`.

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.287, which I just published. It will also be included in a future release of pylance.
